spec/intl.html

```diff
@@ -2210,10 +2213,12 @@ <h1>Temporal.Calendar.prototype.fields ( _fields_ )</h1>
                     1. Let _completion_ be ThrowCompletion(a newly created *RangeError* object).
                     1. Return ? IteratorClose(_iteratorRecord_, _completion_).
                 1. Append _nextValue_ to the end of the List _fieldNames_.
-            1. If _calendar_.[[Identifier]] is *"iso8601"*, then
-              1. Let _result_ be _fieldNames_.
-            1. Else,
-              1. Let _result_ be CalendarDateFields(_calendar_.[[Identifier]], _fieldNames_).
+            1. Let _result_ be _fieldNames_.
+            1. If _calendar_.[[Identifier]] is not *"iso8601"*, then
+              1. NOTE: Every built-in calendar preserves all input field names in output.
+              1. Let _extraFieldDescriptors_ be CalendarFieldDescriptors(_calendar_.[[Identifier]], ~date~).
+              1. For each Calendar Field Descriptor Record _desc_ of _extraFieldDescriptors_, do
+                1. Append _desc_.[[Property]] to _result_.
```

Comment:
@gibson042, me and @ptomato were discussing this change for polyfill changes and we believe, these steps might be slightly incorrect. 
Previously, `calendar.fields(['monthCode', 'day'])` would have returned `['monthCode', 'day']` but with this change it would return `['monthCode', 'day', 'era', 'eraYear']`

We believe there should be a check for determining the _type_ from the input to the fields! 
What do you think?

Reply:
I agree that this PR should not affect the output `builtinNonISOCalendar.fields(['monthCode', 'day'])`, but wasn't happy with the specific changes so I have moved derivation logic back into `CalendarFieldDescriptors` for use only by `Temporal.Calendar.prototype.fields`.

Reply:
This seems like a good solution; keep `~date~`, `~year-month~`, and `~month-day~` for `dateFromFields()`, `yearMonthFromFields()`, and `monthDayFromFields()` respectively; and pass the array directly for `fields()`.

@Aditi-1400 Even though it's a while ago, you might still have more insight into this than I do; want to take a look?